### PR TITLE
Ibp 4209 create inventory lot in search germplasm

### PR DESCRIPTION
**Pull request by lik-83**
Please, review

Include:
IntegratedBreedingPlatform/Middleware#991
IntegratedBreedingPlatform/Fieldbook#1018
IntegratedBreedingPlatform/BMSAPI#548

### CODE REVIEW

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-manager.route.ts [hunk 1]
```diff
@@ -99,5 +99,10 @@ export const GERMPLASM_MANAGER_ROUTES: Routes = [
         path: 'germplasm-import',
         component: GermplasmImportPopupComponent,
         outlet: 'popup'
+    },
+    {
+        path: 'lot-creation',
+        component: LotCreationPopupComponent,
+        outlet: 'popup',
     }
 ];
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-manager.route.ts:108]: You can also place it below `lot-creation-dialog` route, for cohesion

File: src/main/jhipster/src/main/webapp/app/shared/auth/permissions.ts [hunk 2]
```diff
@@ -31,3 +31,11 @@ export const IMPORT_GERMPLASM_UPDATES_PERMISSIONS = [
...
     'IMPORT_GERMPLASM_UPDATES'
 ];
+
+export const CREATE_INVENTORY_LOT_PERMISSIONS = [
+    'ADMIN',
+    'GERMPLASM',
+    'MANAGE_GERMPLASM',
+    'MG_MANAGE_INVENTORY',
+    'MG_CREATE_LOTS'
+];
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/shared/auth/permissions.ts:41]: We can expand it and use it for `lot-creation-dialog` route too

**Review by nahuel-soldevilla — APPROVED** [review]
Looks great :+1: 
Later we can think of renaming or creating a new handler _(onRefresh, onReload)_ for refreshing the table instead of columnFiltersChanged

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-manager.route.ts (hunk 1 above)
Comment (lik-83) [reply src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-manager.route.ts]: Done

File: src/main/java/org/generationcp/breeding/manager/listmanager/listcomponent/ListViewActionMenu.java [hunk 3]
```diff
@@ -192,11 +186,6 @@ protected void layoutAdminLink() {
 		this.codingAndGroupingOptions.addItem(this.messageSource.getMessage(Message.UNGROUP));
 	}
 
-	@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_CROP_MANAGEMENT', 'ROLE_LISTS', 'ROLE_GERMPLASM_LISTS', 'ROLE_MG_MANAGE_INVENTORY', 'ROLE_MG_CREATE_LOTS')")
-	private void addCreateInventoryLotsLink() {
-		this.createInventoryLots = this.addItem(this.messageSource.getMessage(Message.CREATE_INVENTORY_LOTS_MENU_ITEM));
```
Comment (clarysabel) [inline src/main/java/org/generationcp/breeding/manager/listmanager/listcomponent/ListViewActionMenu.java:197]: @lik-83 can the Message CREATE_INVENTORY_LOTS_MENU_ITEM  be removed?

Comment (lik-83) [reply src/main/java/org/generationcp/breeding/manager/listmanager/listcomponent/ListViewActionMenu.java:197]: Done!

**Review by nahuel-soldevilla — APPROVED** [review]
Looks good though some things we'll probably don't need in the future, like the `openedFromWorkbench` condition. But this should be fine for now.

The simplified form would involve (considering a gradual transition Fieldbook -> Workbench involving iframes):
- Headers always inside iframe
- No need for inline alerts (jhiAlertService.addAlert toast=false), if there's an iframe, the alert popup would just be displayed inside the iframe. Sometimes there are long (with scroll) modals, and for those cases the popup is not visible. In those cases we can stick with just inline alerts until a full migration, and then switch to toast.
- call parent iframe close handler ([either through callback variables or events](https://github.com/IntegratedBreedingPlatform/Workbench/blob/c6c74d3fdcbf0b7042e910fbd51101566e97adc8/src/main/jhipster/src/main/webapp/app/germplasm-manager/selector/germplasm-selector.component.ts#L559) + ngbootstrap activeModal close

**Pull request merged**